infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -436,6 +436,7 @@ private void onLeadershipAcquire() {
                 // Standby Site : the LogReplicationServer (server handler) will initiate the LogReplicationSinkManager
                 log.info("Start as Sink (receiver)");
                 interClusterReplicationService.getLogReplicationServer().setLeadership(true);
+                interClusterReplicationService.getLogReplicationServer().getSinkManager().reset();
```

Comment:
should this rather be reset on negotiation? because imagine the case where you lost the lock for a short period or even a long period but writes did not progress, so your lastProcessSequence is still valid as you can continue from that point onwards?

Reply:
Do you mean lost the lock for a short period and there is no negotiation during this period? So we don't need to reset in this case.

Reply:
I see your point. Active won't know remote leader loss if there is no write... Moved it to negotiation